packages/trimerge-sync-indexed-db/src/trimerge-indexed-db.ts

```diff
@@ -240,16 +232,15 @@ class IndexedDbBackend<
       }
     }
     for (const ref of headsToAdd) {
-      promises.push(headsDb.add({ ref }));
+      promises.push(headsDb.put({ ref }));
     }
 
     if (lastSyncId) {
       await this.updateLastRemoteSyncId(lastSyncId);
     }
-    if (promises.length > 0) {
-      await Promise.all(promises);
-      await tx.done;
-    }
+
+    await Promise.all(promises);
+    await tx.done;
```

Comment:
this fixes the transaction AbortError (I was never resolving the transaction if promises.length === 0)